recipe/gen_patch_json.py

```diff
@@ -795,8 +795,27 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
         if record_name == "airflow-with-async":
             _rename_dependency(fn, record, "evenlet", "eventlet")
 
+        # Current implementation in iris allows for frequent SegFaults
+        # when netCDF4>1.6.0; iris devs are working on a solution and
+        # discussing it in https://github.com/SciTools/iris/issues/5016
+        # but a wide repodata patch is needed before a permanent solution
+        iris_deps = [
+            "netcdf4 <1.6.1",
+        ]
+        iris_updates = {
+            "3.1.0": iris_deps,
+            "3.2.0": iris_deps,
+            "3.2.0.post0": iris_deps,
+            "3.2.1": iris_deps,
+            "3.2.1.post0": iris_deps,
+            "3.3.0": iris_deps,
+            "3.3.1": iris_deps,
+            "3.4.0": iris_deps,
+        }
         if record_name == "iris":
             _rename_dependency(fn, record, "nc_time_axis", "nc-time-axis")
+            if record["version"] in iris_updates:
+                record["depends"] = iris_updates[record["version"]]
```

Comment:
This overwrites all of the dependencies. I think you want to append or edit them, right?

Reply:
does it? Yeah all I want is to pin netCDF4 <1.6.1 for all those listed versions. But looking at the show_diff output and the test it looks to me that the deps and their pins are preserved as they are for each version, and only netCDF4 <1.6.1 is appended at the end of the list?

Reply:
Not it removes everything. Notice the `-` on the diff

```diff
noarch::iris-3.3.1-pyhd8ed1ab_0.tar.bz2
-    "cartopy >=0.21",
-    "cf-units >=3.1",
-    "cftime >=1.5",
-    "dask-core >=2",
-    "matplotlib-base",
-    "netcdf4 !=1.6.1",
-    "numpy >=1.19",
-    "pyproj",
-    "python >=3.8",
-    "python-xxhash",
-    "scipy",
-    "shapely !=1.8.3"
+    "netcdf4 <1.6.1"
```

Reply:
crap! Nevermind me, I am a total doofus, I only now noticed the output is an actual diff, fixing it now, thanks a lot @beckermr

Reply:
totally! I should pay attention to signs at the start of the line :laughing: It now should do what's written on the tin, let me upload the diff output to the PR description again